client/components/infinite-list/index.jsx

```diff
@@ -109,19 +110,20 @@ export default class InfiniteList extends Component {
 		// current callstack is executed. Some streams and device widths for the reader are not
 		// fully ready to have their scroll position set until everything is mounted, causing the
 		// stream to jump back to the top when coming back to view from a post.
-		if ( this._contextLoaded() ) {
-			// Use the scrollTop setting from the time the component mounted, as this state could be
-			// changed in the initial update cycle to save scroll position before the saved position
-			// is set.
-			const scrollTop = this.state.scrollTop;
-			// Apply these at the end of the callstack to ensure the scroll container is ready.
-			window.setTimeout( () => {
+
+		// Use the scrollTop setting from the time the component mounted, as this state could be
+		// changed in the initial update cycle to save scroll position before the saved position
+		// is set.
+		const scrollTop = this.state.scrollTop;
+		// Apply these at the end of the callstack to ensure the scroll container is ready.
+		window.setTimeout( () => {
+			if ( this._contextLoaded() ) {
 				this._setContainerY( scrollTop );
 				this.updateScroll( {
 					triggeredByScroll: false,
 				} );
-			}, 0 );
-		}
+			}
+		}, 0 );
```

Comment:
I added this in a recent PR. I just changed it slightly so it evaluates `if ( this._contextLoaded() )` at the time the timeout code runs as opposed to before setting it. It seems more ideal, but I did not notice any behavioral change.